packages/tbas/src/tbas.ts

```diff
@@ -0,0 +1,249 @@
+  const address    = settings.wallet
+  const contractId = tokenContract
+  const tokenId    = uuidToUint256(entityId).toString()
+  console.log('MINTER', address)
+  console.log('CONTRACT', contractId)
+  console.log('ENTITY', entityId)
+  console.log('TOKEN', tokenId)
+
+  if (!address || !contractId || !walletSeed) {
+    throw new Error("Missing wallet or contract info")
+  }
+
+  const response = await BlockchainManager.xdc.server.mintNFT721({
+    address,
+    tokenId,
+    contractId,
+    walletSeed,
+  })
+  if ("error" in response) {
+    throw new Error(response.error)
+  }
+  console.log("Minted NFT", response.txId, response.tokenId)
+  return response
+}
+
+// Create token bound account from implementation
+export async function createAccount(tokenContract:string, tokenId:string, chainId:string, waitForReceipt=false){
+  console.log('Creating account...')
+  const privateKey = process.env.XDC_WALLET_SECRET || ''
+  if(!privateKey){
+    return { status: 'error', txid:'', address:'', error:'Private key not found' }
+  }
+  try {
+    const client = newClient()
+    const serverAccount = privateKeyToAccount(privateKey as Address)
+    const contract = getContract({
+      client,
+      abi: registryABI,
+      address: registryAddress
+    })
+
+    // Simulate
+    const { request } = await client.public.simulateContract({
+      account: serverAccount,
+      address: registryAddress,
+      abi: registryABI,
+      functionName: 'createAccount',
+      args: [implementationAddress, baseSalt, BigInt(chainId), tokenContract as Address, BigInt(tokenId)],
+    })
+
+    // Send to chain
+    const txid = await client.wallet.writeContract(request)
+    console.log('TXID', txid)
+
+    // Tx receipt
+    if(waitForReceipt){
+      const receipt = await getReceipt(txid)
+      if(!receipt){
+        return { status: 'notfound', txid, address:'', error:'' }
+      }
+      if(receipt.status==='success'){
+        const address = `0x${receipt?.logs[0].data.substr(26,40)}`
+        console.log('TBA', address, receipt.status)
+        return { status: receipt.status, txid, address, error:'' }
+      }
+      return { status: receipt.status, txid, address:'', error:'' }
+    }
+    return { status: 'pending', txid, address:'', error:'' }
+  // biome-ignore lint/suspicious/noExplicitAny: any error
+  } catch(ex:any) {
+    console.error(ex)
+    return { status: 'error', txid:'', address:'', error:ex.message }
+  }
+}
+
+// Get account address from contract
+export async function getAccount(tokenContract:string, tokenId:string, chainId:string){
+  console.log('Getting account...', tokenContract, tokenId, chainId)
+  const client = newClient()
+  const contract = getContract({
+    client: client.public,
+    abi: registryABI,
+    address: registryAddress
+  })
+  const address = await contract.read.account([
+    implementationAddress,
+    baseSalt,
+    BigInt(chainId),
+    tokenContract as Address,
+    BigInt(tokenId)
```

Comment:
does this need to use the uuidToUint256 function?

Reply:
Yes as there is no way to pass a uuid as int to contracts

Reply:
Because tokenId already has been converted to int from uuid in line 129, now we pass it as bigint as contracts use int256 instead of string or ints